shoulder/generator/c_header_generator.py

```diff
@@ -42,15 +42,13 @@ def generate(self, objects, outpath):
             regs = transforms["remove_reserved_sign_extended"].transform(regs)
             regs = transforms["remove_implementation_defined"].transform(regs)
             regs = transforms["special_to_underscore"].transform(regs)
-            regs = transforms["remove_coprocessor_am"].transform(regs)
-            regs = transforms["remove_memory_mapped_am"].transform(regs)
-            regs = transforms["remove_system_vector_am"].transform(regs)
-            regs = transforms["remove_system_banked_am"].transform(regs)
-            regs = transforms["remove_system_immediate_am"].transform(regs)
```

Comment:
Do the AArch32 and External filters cover all the unimplemented methods that are represented by these filters?

Reply:
Yes. I checked all of the register files in the manuals and verified that AArch32 == [mrc, mrrc, mcr, mcrr, vmsr, vmrs, mrs_banked, msr_banked] access mechanisms in *all* cases, and External == [ldr, str] access mechanisms in *all* cases.

Therefore, applying the AArch32 and External filters should give the *exact* same result as using all the transforms above.

Reply:
After thinking about it, it's not quite *exactly* the same because the msr_immediate access mechanisms won't be filtered out. However, the generator prefers msr_register over msr_immediate, so the generated header will be the same.

Reply:
:+1: